crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1466,26 +1463,11 @@ impl Client {
             versions.push(MatrixVersion::V1_0);
         }
 
-        let unstable_features = resp.unstable_features;
```

Comment:
Why is it better to inline this in the caller method?

Reply:
I don't understand this comment?

Reply:
Err, I missed a bit, I wanted to point at this piece of code: https://github.com/matrix-org/matrix-rust-sdk/pull/3658/commits/7917bbb7c6ded8b1ad7b402438fb2aa94560b4e1#diff-6d2ff673b14b2a563388b2d177b0c4e7ff2c16714054858fdec9bfd01858ba96L1471-L1484.

The caching of the result was moved out of this method, and I was wondering why that is considered to be better.

Reply:
I found that this layered responsibilities slightly better:

- one method is only concerned about network
- one method is only concerned about storage, and will resort to network if needs be, then cache in storage
- one method is only concerned about memory caching, and will resort to storage etc.